File: python/weather/weather.py
import requests

def get_weather(city):
    api_key = "YOUR_API_KEY"
    url = f"http://api.openweathermap.org/data/2.5/weather?q={city}&appid={api_key}&units=metric"
    response = requests.get(url)
    data = response.json()
    return data

def main():
    print("Welcome to the Simple Weather App!")
    city = input("Enter city name: ")

    try:
        weather_data = get_weather(city)
        if weather_data["cod"] == 200:
            print("\nCurrent Weather in", city)
            print("Description:", weather_data["weather"][0]["desription"])
            print("Temperature:", weather_data["main"]["temp"], "°C")
            print("Humidity:", weather_data["main"]["humidity"], "%")
            print("Wind Speed:", weather_data["wind"]["speed"], "m/s")
        else:
            print("City not found. Please enter a valid city name.")
    except Exception as e:
        print("An error occurred:", e)

if __name__ == "__main__":
    main()

File: typescript/weather/weather.ts
#!/usr/bin/env node

import axios from 'axios';

async function getWeather(city: string): Promise<void> {
	const apiKey = 'YOUR_API_KEY';
	const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;

		try {
		const response = await axios.get(url);
		const { main, weather } = response.data;
		const temperature = main.temp;
		const weatherDescription = weather[0].description;

		console.log(`Current weather in ${city}:`);
		console.log(`Temperature: ${temperature}°C`);
		console.log(`Description ${weatherDescription}`);
	} catch (error) {
		console.error(`Error fetching weather data: ${error.message}`);
	}
}

function main(): void {
	const city = process.argv[2];
	if (!city) {
		console.log("Please provide a city name as an argument");
		return;
	}

	getWeather(city);
}

main();
